src/compiler/program.ts

```diff
@@ -1757,9 +1757,17 @@ namespace ts {
                 return flatDiagnostics;
             }
 
+            const isCheckJs = isCheckJsEnabledForFile(sourceFile, options);

```

Comment:
You should instead just pass on `includeBindAndCheckDiagnostics` and change above condition to 
```ts
if(!includeBindAndCheckDiagnostics || !sourceFile.commentDirectives?.length) {
  return flatDiagnostics;
 }
```

Reply:
changed, thanks